### PR TITLE
reorg travis, move to xonsh based scripting, and template

### PR DESCRIPTION
**Pull request by CJ-Wright**
``.travis.yml`` files via jinja

Attn: @scopatz

### CODE REVIEW

**Comment by CJ-Wright** [comment]
Attn: @justcalamari

File: configure_workers.xsh [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+                                     'script_number': '02'},
+                  'make-cf-graph': {'doctr_key': 'kUlHJeVjxNf/dt9XSSV+mgHH/9qp30zqABjwdeDe1Nbs5JPVCuLjD7yqL/Pln7jQdzGaTksEIUoFggGSKpXi1SCIZCOlzBax/wAIPTSvJEvUIzFs6Vws7J/mV147tyoNanOHvNw6CvicNhEuWD8HCWUw36kF93yhePwuFvNRWgIof8Ua54d3l2sGTCEHSbIiSvx6eF5WVsIKOWVyjKGu5H3y6HSoOGkFlezw+d9Y2owkx99KPKgDRFRt1BCuKxjhi8Yuv5HTKzCwDuQFC6Ak8uriVTQfMv44Tssl3SP4bDu+1x+zr7mMHTTOHqlYFSOlBtSkDAf6Cctfk903kFQI5TkiDLCSe86tlUiiC2qcFtgL4kll4SMQ5Eu1nmoCjQk457KUb5nkIpWFS5g/TxM8qy0CZF/KHsz2n9M76iDg8V2qS5xyb2hglt51HCGb94tnuR6oBPx+yDzBTGm38c4I8vpzT+8uDEr9bbjh7pO0hzssNCSc/XZld8ctcA5l1M+9Ib/W7QwqK0uelh2VEwpnd8JeMEXGw5jaT+1EA3FUnNTBBOGzJXK3SzG8LAJdQjChSl8vdDvuug2VN04433oZ+PhWv9YrPc+XBs++X/re4pYK7t1dXFBx7qlyMFRWTCYhiPhzgBwbeojuB/MuL6jGld9UqMews+NjdFIyY3j0W6o=',
+                                    'script_number': '03'}}
+
+r = '/home/christopher/dev/'
```
Comment (justcalamari) [inline configure_workers.xsh]: We might want to make this a command line argument, or perhaps an environment variable.

File: 00.xsh [hunk 2]
```diff
@@ -1,10 +1,8 @@
-#!/usr/bin/env bash
-
-export START_TIME="$(date +%s)"
-export TIMEOUT=2700
-./setup.sh
+$START_TIME = int("$(date +%s)")
+$TIMEOUT = 2700
+sh setup.sh
```
Comment (justcalamari) [inline 00.xsh]: You run setup.sh here, but also in travis_template.yml.

File: 00.xsh [hunk 3]
```diff
@@ -1,10 +1,8 @@
-#!/usr/bin/env bash
```
Comment (scopatz) [inline 00.xsh]: This should be `#!/usr/bin/env xonsh`

Comment (CJ-Wright) [reply 00.xsh]: (Or just remove it)

Comment (scopatz) [reply 00.xsh]: And likewise for all of the scripts

File: 00.xsh (hunk 2 above)
Comment (scopatz) [reply 00.xsh]: Also do you want to run this or `source-bash` it?

File: configure_workers.xsh [hunk 4]
```diff
@@ -0,0 +1,36 @@
+"""Script for configuring autotick workers based off of a configuration and
```
Comment (scopatz) [inline configure_workers.xsh]: Should also have a shebang line, right?

File: configure_workers.xsh [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+    [os.path.expanduser('.')]))
+
+for k, ctx in configurations.items():
+    d = os.path.join($BOT_ROOT, k)
+    output = os.path.join(d, '.travis.yml')
+    template = env.get_template('travis_template.yml')
+    result = template.render(ctx)
+    with indir(d):
+        with open(output, 'w') as f:
+            print(f.name)
```
Comment (scopatz) [inline configure_workers.xsh]: This print does not need to be done while the file is open

**Comment by CJ-Wright** [comment]
Thank you!

**Pull request merged**